src/Main.js

```diff
@@ -0,0 +1,189 @@
+import React, { Component } from 'react'
+import ProfilesPage from './ProfilesPage.js'
+import LandingPage from './LandingPage.js'
+import LogInPage from './LogInPage.js'
+import SignUp from './SignUp.js'
+import firebase from './firebase.js'
+
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+
+import * as actions from './actions.js'
+
+const LANDING = 'landing'
+const LOG_IN = 'login'
+const SIGN_UP = 'signup'
+const PROFILES = 'profiles'
+
+class Main extends Component {
+  constructor (props) {
+    super(props)
+    this.handleSubmit = this.handleSubmit.bind(this)
+    this.logIn = this.logIn.bind(this)
+    this.setUser = this.setUser.bind(this)
+    this.clearUser = this.clearUser.bind(this)
+    this.handleChange = this.handleChange.bind(this)
+  }
+
+  logIn (e) {
+    e.preventDefault()
+    let userUID
+    const setUser = this.setUser
+    firebase.auth().signInWithEmailAndPassword(this.props.email, this.props.password)
+      .then(() => {
+        firebase.auth().onAuthStateChanged((fbuser) => {
+          if (fbuser) {
+            userUID = fbuser.uid
+            setUser(userUID)
+          } else {
+            userUID = null
+            setUser(userUID)
+          }
+        })
+      })
+      .catch(function (error) {
+        const errorCode = error.code
+        const errorMessage = error.message
+        console.log('catch ran on logIn', errorCode, errorMessage)
+      })
+  }
+
+  setUser (userUID) {
```

Comment:
This method seems like it belongs in the reducer to me. I think you're having `Main.js` do more state-management than it needs to.

Reply:
Never mind! don't do this.